### DISCUSSION TITLE
request.getfixturevalue in pytest==7.1.x

### DISCUSSION THREAD

**Discussion by Armantic** [pytest-dev/pytest#9812] [Q&A]
Hi!

Could you please help me?
I have code:
```
import pytest
from _pytest.fixtures import FixtureRequest

@pytest.hookimpl(hookwrapper=True)
def pytest_runtest_setup(item: pytest.Function):
    request = FixtureRequest(item)
    tmpdir = request.getfixturevalue('tmpdir').strpath
```
and I want to get value of fixture 'tmpdir' in pytest_runtest_setup hook. 
For pytest==6.x.x it works, but not on 7.1.x (FixtureLookupError raises on request.getfixturevalue('tmpdir')). 
Is it a bug? Or how can I get the value of 'tmpdir' in pytest_runtest_setup in another way?

**Comment by RonnyPfannschmidt** [comment]
Never under any circumstances should you create a own fixture request object distinct from the one pytest manages

What's the surrounding usecase?

**Reply by Armantic** [reply]
@RonnyPfannschmidt 
How can I get the value of 'tmpdir' (or other fixtures) in pytest_runtest_setup in another way?

"What's the surrounding usecase?"
For example, I want to use 'tmpdir' for preparing some sources (downloading or copying some files to 'tmdir' before running the test)
I just don't want to use autouse fixture (which will accept the fixture 'tmpdir') for this

**Reply by RonnyPfannschmidt** [reply]
for the presented use-case - a autouse fixture is the intended way to do it, 
its certainly absolutely incorrect to create a different unrelated fixturerequest to fill a additional fixture

its safe to say that your code worked by accident and is entirely unsupported

**Reply by Armantic** [reply]
ok, I got it, thank you
